gradio/components.py

```diff
@@ -335,6 +340,41 @@ def base64_to_temp_file_if_needed(
         self.temp_files.add(full_temp_file_path)
         return full_temp_file_path
 
+    def pil_to_temp_file(self, img: _Image.Image, dir: str, format=".png") -> str:
+        temp_dir = Path(dir) / self.hash_bytes(img.tobytes())
+        temp_dir.mkdir(exist_ok=True, parents=True)
+        filename = str(temp_dir / f"image.{format}")
+        img.save(filename, pnginfo=processing_utils.get_pil_metadata(img))
+        return filename
+
+    def img_array_to_temp_file(self, arr: np.ndarray, dir: str) -> str:
+        temp_dir = Path(dir) / self.hash_bytes(arr.tobytes())
+        temp_dir.mkdir(exist_ok=True, parents=True)
+        filename = str(temp_dir / "image.png")
+        pil_image = _Image.fromarray(
+            processing_utils._convert(arr, np.uint8, force_copy=False)
+        )
+        pil_image.save(filename)
+        return filename
```

Comment:
Why not
```suggestion
    def img_array_to_temp_file(self, arr: np.ndarray, dir: str, format=".png") -> str:
        pil_image = _Image.fromarray(
            processing_utils._convert(arr, np.uint8, force_copy=False)
        )
        return self.pil_to_temp_file(pil_image, dir=dir, format=format)
```
?
That would ensure a NumPy array describing an image and the equivalent PIL image get the same hash...

Reply:
Good point! Added a test for this too